containers.md

```diff
@@ -0,0 +1,96 @@
+
+Minikube also sets up a Development Kubernetes cluster that you can use for learning, or as an alternative to docker-compose
+
+## Building a Container Image
+
+### Use small base images
+
+Theres thousands of base images you could choose from when building an image. Finding a good base image is one of the most important parts of your building your own image. Ideally your base image should include as little as possible, just the bare minimum to run your app, this helps reduce your final image size and also reduce the potential attack surface.
+
+ - [Alpine](https://hub.docker.com/_/alpine) is a very commonly used base image. Its extremely small image just being 5MB in size. By itself it probably doesn't have what you need, but you'll find that popular images tend to publish Alpine versions. [Node for example](https://hub.docker.com/_/node)adds the `-alpine` suffix to any images that use an Alpine base.
+ - [Distroless](https://github.com/GoogleContainerTools/distroless) Google maintains a number of images that are completely stripped down to what is absolutely necessary, they are even smaller than Alpine images. This can mean they don't even include standard system libraries that we take for granted such as libssl!
+
+Alpine is a good start for building small images, its not quite as small as Distroless but its a lot easier to work with.
+
+### Avoid packaging intermediate files / code / build tools into your final Image
+
+Similar to Git, any files that get added to an image will remain in its history and can be restored even if they're deleted before the end of the build. This is to consider when optimizing your images for size as including something such as a `node_modules` folder before it has been bundled can quickly balloon your image size to be orders of magnitude larger than it needs to be.
+
+Theres 2 ways you can avoid this issue:
+  - **A**. Build your application outside of the Image build environment, and then copy your final build artifacts into the image.
+  - **B**. [Use a multi-stage build](https://docs.docker.com/build/building/multi-stage/), files added in a previous build stage won't be included in the next build stage unless specified.
+
+### Use `buildah` to compile images
+
+Buildah is another open source tool developed by Redhat, it produces OCI compliant images meaning they can generally be run in both Docker and Podman. 
+
+You may also want to consider renaming `Dockerfiles` to `Containerfiles`. 
+
+## Publishing a Container Image
+
+Instead of using the Docker container registry we recommend using [GitHubs built in Container Registry](https://docs.github.com/en/packages/working-with-a-github-packages-registry/working-with-the-container-registry). This registry is free to use and has no limits for any public repositories and also integrates extremely well with Github Actions.
+
+Below is an example Github workflow which handles the build and publishing of a container image to Github Container Registry.
+
+```yaml
```

Comment:
Would it be better to link to an exaple? That way we might be more confident the example is being kept up to date?

For some of these actions we might want to use a commit hash, rather than the tag?